src/feed/views/funding_feed_view.py

```diff
@@ -79,140 +79,132 @@ def list(self, request, *args, **kwargs):
+        
+        return self._apply_ordering(queryset, ordering, fundraise_status)
+
+    _fundraise_content_type = None
+    
+    @classmethod
+    def _get_fundraise_content_type(cls):
+        if cls._fundraise_content_type is None:
+            cls._fundraise_content_type = ContentType.objects.get_for_model(Fundraise)
+        return cls._fundraise_content_type
+    
+    def _build_base_queryset(self, created_by=None):
+        now = timezone.now()
+        
+        has_active = Fundraise.objects.filter(
+            unified_document_id=OuterRef("unified_document_id"),
+            status=Fundraise.OPEN
+        ).filter(Q(end_date__isnull=True) | Q(end_date__gt=now)).values('pk')[:1]
+        
+        contributor_count_subquery = Purchase.objects.filter(
+            content_type=self._get_fundraise_content_type(),
+            object_id=OuterRef("pk")
+        ).values('object_id').annotate(
+            count=Count('user_id', distinct=True)
+        ).values('count')
+        
+        fundraise_prefetch = Prefetch(
+            "unified_document__fundraises",
+            queryset=Fundraise.objects.select_related("escrow").prefetch_related(
+                "nonprofit_links"
+            ).annotate(
+                contributor_count=Coalesce(
+                    Subquery(contributor_count_subquery, output_field=IntegerField()),
+                    0
                 )
-                # Order by end date descending (most recent deadlines first)
-                queryset = queryset.order_by("-unified_document__fundraises__end_date")
-        else:
-            # For ALL tab: We need different sorting for OPEN vs CLOSED/COMPLETED
-            # Sort first by status (OPEN first), then apply different date sorts
-            # based on status
-            queryset = queryset.annotate(
-                # Create a flag to identify OPEN fundraises
-                is_open=Case(
-                    When(
-                        unified_document__fundraises__status=Fundraise.OPEN,
-                        then=Value(True),
-                    ),
-                    default=Value(False),
-                    output_field=BooleanField(),
-                ),
-            ).order_by(
-                "-is_open",
-                # For OPEN (is_open=True): Sort by closest (earliest) end_date first
-                Case(
-                    When(
-                        is_open=True, then=F("unified_document__fundraises__end_date")
-                    ),
-                ),
-                # For CLOSED (is_open=False): Sort by most recent end_date first
-                Case(
-                    When(
-                        is_open=False, then=F("unified_document__fundraises__end_date")
-                    ),
-                    default=None,
-                ).desc(),
+            ).order_by("end_date")
+        )
+        
+        queryset = ResearchhubPost.objects.select_related(
```

Comment:
We should probably annotate this query with some comments especially the "annotate" clause because it is hard to understand what this does at a glance